### PR TITLE
cleanup

### PR DESCRIPTION
**Pull request by owlen**
Some cleanup while going over the code, trying to figure it out...

### CODE REVIEW

**Comment by Schwarzbaer** [comment]
> Of the entity's :class:Model. I'm still wrapping my own head around it, but an entity may have many root nodes in the scene graph, since different mechanics that it partakes in may be completely unrelated to each other.

> Ok, I'm not sure I follow you completely. Still struggling with the all the different classes of the different systems involved (panda3d/wecs etc). Can you give an example for this? I have a feeling this is something that have to be documented somehow.

You wrote that the movement system moves the entity. But the entity has no single body to move. It could have a `Model`, a character (once that doesn't use `Model` anymore), effects affecting areas or characters or objects in the game world... And each of that could operate in the scene graph independent of all the others. So that's why not the entity is moved, but the entity's `Model`.

File: docs/manual/contribute.md [hunk 1]
```diff
@@ -0,0 +1,28 @@
+# Contributing to WECS
+
+WECS is in very early dev phase, and we'd love an ysupport we can get.
```
Comment (janEntikan) [inline docs/manual/contribute.md]: "we'd love any support" instead of "we'd love an ysupport"

File: docs/manual/contribute.md [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+## Coding
+
+[comment]: # (TODO this should be expanded, and maybe the TOD should be merge here somehow)
+
+Feel fre to fork the project, play and expand upon it. Have a look at the TODO page.
```
Comment (janEntikan) [inline docs/manual/contribute.md]: "feel free" instead of "feel fre"

**Comment by Schwarzbaer** [comment]
I'm still working on getting an overview, but a few things in the file list stuck out.

```
requirements.txt
```
~~To install `wecs`, you first need to install `wecs`? Wat?
Also I *think* we can drop `recommonmark`; IIRC that was the attempt to get Sphinx to process Markdown by itself, and I ripped everything about it out again after changing to pandoc.~~

```
docs/manual/contribute.rst
```
~~You didn't add that to `docs/manual/manual.rst`, so it won't show up in the manual. I think right before or after the design notes would be a good place.~~

```
docs/manual/design.rst
```
I assume that you edited this before knowing of `gen.sh`? Since `design.rst` is generated from `design.md`, these changes will simply be overwritten when running `gen.sh`. I'm wondering why they aren't being overwritten for you though...

```
examples/panda3d-pong/resources/ball.bam
examples/panda3d-pong/resources/ball.blend
examples/panda3d-pong/resources/paddle.bam
examples/panda3d-pong/resources/paddle.blend
```
~~Why were these changed? Pong still looks the same to me.~~ (Files were moved to resource folder)

```
wecs/templates/entity.html
```
~~You added `<!--suppress JSUnusedGlobalSymbols -->`; I assume that has some special meaning to some tool?~~

**Comment by Schwarzbaer** [comment]
`examples/minimal/main.py`, two empty lines between imports and code, please.

**Review by janEntikan — APPROVED** [review]
Looks good to me. Let's merge this before moving on.

**Comment by owlen** [comment]
> > Of the entity's :class:Model. I'm still wrapping my own head around it, but an entity may have many root nodes in the scene graph, since different mechanics that it partakes in may be completely unrelated to each other.
> 
> > Ok, I'm not sure I follow you completely. Still struggling with the all the different classes of the different systems involved (panda3d/wecs etc). Can you give an example for this? I have a feeling this is something that have to be documented somehow.
> 
> You wrote that the movement system moves the entity. But the entity has no single body to move. It could have a `Model`, a character (once that doesn't use `Model` anymore), effects affecting areas or characters or objects in the game world... And each of that could operate in the scene graph independent of all the others. So that's why not the entity is moved, but the entity's `Model`.

You're not only, right, I also understand it much better now. I'll update with some rewritting.

**Comment by owlen** [comment]
> I'm still working on getting an overview, but a few things in the file list stuck out.
> 
> ```
> requirements.txt
> ```
> 
> To install `wecs`, you first need to install `wecs`? Wat?
> Also I _think_ we can drop `recommonmark`; IIRC that was the attempt to get Sphinx to process Markdown by itself, and I ripped everything about it out again after changing to pandoc.
> 
> ```
> docs/manual/contribute.rst
> ```
> 
> ~You didn't add that to `docs/manual/manual.rst`, so it won't show up in the manual. I think right before or after the design notes would be a good place.~
> 
> ```
> docs/manual/design.rst
> ```
> 
> I assume that you edited this before knowing of `gen.sh`? Since `design.rst` is generated from `design.md`, these changes will simply be overwritten when running `gen.sh`. I'm wondering why they aren't being overwritten for you though...
> 
> ```
> examples/panda3d-pong/resources/ball.bam
> examples/panda3d-pong/resources/ball.blend
> examples/panda3d-pong/resources/paddle.bam
> examples/panda3d-pong/resources/paddle.blend
> ```
> 
> Why were these changed? Pong still looks the same to me.
> 
> ```
> wecs/templates/entity.html
> ```
> 
> You added `<!--suppress JSUnusedGlobalSymbols -->`; I assume that has some special meaning to some tool?

ּRemoved. Also the other comments inside are handled in the last commits.

**Comment by Schwarzbaer** [comment]
Oof... That was a beast of a PR to review, so I tip my hat to you for powering through creating it. Here's my notes dump~~, I still have to comb through the discussion above to weed out duplicates~~.

* README.md: ~~Discord invite link is missing; https://discord.gg/hXGyd43~~
* wecs/aspects.py: ~~`Aspect.__call__`: That's explained in the manual.~~
* wecs/boilerplate.py:
  * ~~Only one import per line, please.~~
  * FIXME in `if console:`: It's not redundant. `debug_keys` use Panda3D events. The console may exist without them being bound, and use e.g. `wecs.panda3d.debug` to toggle the console instead. Doing both at the same time would be bad.
  * fixme in `# Set up the world:`: ~~It's in the manual that `game` is used. We should make that a config value though.~~
* wecs/core.py:
  * `World.get_entity`: ~~"world entity"~~
  * `World.add_system`: ~~You cut of the end of the last sentence.~~
  * `OrFilter`: ~~Private? Protected? In Python?~~
* wecs/mechanics/clock.py: ~~Please don't use TODO in documentation. Rather file an issue.~~
* examples/panda3d-character-controller/game.py:
  * `system_types`: `wp3dAcceptInput` is missing a `.`.
  * The fixme: Yeah, I was a bit overeager to add that comment to boilerplate. I'll deal with it... eventually. For the moment, the fixme in `boilerplate.py` is enough though.
* examples/panda3d-pong/movement.py:
  * ~~Most of the module docstring is redundant, since you're repeating it in the API docs of the classes, which appears right after in the rendered docs anyway; So it should be cut. In general, when writing module docstrings, please add which components / system you mean in the module string. "The :class:`Movement` component holds..."~~
  * `MoveObject.update`: ~~To finish with the smallest niggle, there's an extra space of indentation in the docstring in the last line before params. :D~~

File: docs/manual/contribute.md (hunk 1 above)
Comment (Schwarzbaer) [reply docs/manual/contribute.md]: Updated.

**Comment by owlen** [comment]
Besides not finding a todo in clock.py, I think I address all in the last commit.

**Comment by Schwarzbaer** [comment]
> Besides not finding a todo in clock.py, I think I address all in the last commit.

In the module docstring; `""" TODO add description of what mechanics.clock does"""`

**Pull request merged**